### PR TITLE
feat: add documentation for new types exported from `astro:content`

### PR DESCRIPTION
**Pull request by Princesseuh**
#### What kind of changes does this PR include?

- New or updated content

#### Description

Adds documentation for https://github.com/withastro/astro/pull/7607.

This PR also adds a mention of the `SchemaContext` type that already exists

### CODE REVIEW

File: src/content/docs/en/reference/api-reference.mdx [hunk 1]
```diff
@@ -1028,6 +1028,57 @@ const { Content, headings, remarkPluginFrontmatter } = await entry.render();
...
+}
+```
+
+#### `ContentCollectionKey`
+
+An union of string of all the names of `type: 'content'` collections defined in your `src/content/config.*` file.
+
+#### `DataCollectionKey`
+
+An union of string of all the names of `type: 'data'` collection defined in your `src/content/config.*` file.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1053]:
```suggestion
#### `ContentCollectionKey`

A string union of all the names of `type: 'content'` collections defined in your `src/content/config.*` file.

#### `DataCollectionKey`

A string union of all the names of `type: 'data'` collection defined in your `src/content/config.*` file.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 2]
```diff
@@ -1028,6 +1028,57 @@ const { Content, headings, remarkPluginFrontmatter } = await entry.render();
...
+import type { CollectionKey, getCollection } from 'astro:content';
+
+async function getCollection(collection: CollectionKey) {
+  return getCollection(collection);
+}
+```
+
+#### `ContentCollectionKey`
+
+An union of string of all the names of `type: 'content'` collections defined in your `src/content/config.*` file.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1049]:
```suggestion
A string union of all the names of `type: 'content'` collections defined in your `src/content/config.*` file.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 3]
```diff
@@ -1028,6 +1028,57 @@ const { Content, headings, remarkPluginFrontmatter } = await entry.render();
 
 [See the `Content Collection` guide](/en/guides/content-collections/#rendering-content-to-html) for example usage.
 
+### Other Content Collection Types
+
+The `astro:content` module also exports the following types for use in your Astro project:
+
+#### `CollectionKey`
+
+An union of string of all collection names defined in your `src/content/config.*` file. This type can be useful when defining a generic function that accepts any collection name.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1037]:
```suggestion
A string union of all collection names defined in your `src/content/config.*` file. This type can be useful when defining a generic function that accepts any collection name.
```

File: src/content/docs/en/reference/api-reference.mdx [hunk 4]
```diff
@@ -1028,6 +1028,57 @@ const { Content, headings, remarkPluginFrontmatter } = await entry.render();
...
+
+#### `DataCollectionKey`
+
+An union of string of all the names of `type: 'data'` collection defined in your `src/content/config.*` file.
+
+#### `SchemaContext`
+
+The `context` object that `defineCollection` uses for the function shape of `schema`. This type can be useful when building reusable schemas for multiple collections.
+
+This includes the following properties:
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1059]:
```suggestion
This includes the following property:
```
If there's only one property so far?

File: src/content/docs/en/reference/api-reference.mdx [hunk 5]
```diff
@@ -1028,6 +1028,57 @@ const { Content, headings, remarkPluginFrontmatter } = await entry.render();
...
+
+- `image` - The `image()` schema helper
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1061]:
```suggestion
- `image` - The `image()` schema helper that... (allows you to.../ enforces.../types.../defines...)
```
I feel like this maybe wants a bit more here?

**Review by sarah11918 — COMMENTED** [review]
Got those string unions! Also just a couple of tiny possible suggestions. Otherwise, good to go!

File: src/content/docs/en/reference/api-reference.mdx (hunk 5 above)
Comment (Princesseuh) [reply src/content/docs/en/reference/api-reference.mdx:1061]:
```suggestion
- `image` - The `image()` schema helper that allows you [to use local images in Content Collections](/en/guides/images/#images-in-content-collections)
```

**Pull request merged**